File: medplum-java/medplum-model/src/main/java/com/fhir/MedicinalProduct_NamePart.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMedicinalProduct_NamePart.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MedicinalProduct_NamePart {
  Optional<String> part();

  Optional<List<Extension>> modifierExtension();

  Coding type();

  Optional<String> id();

  Optional<List<Extension>> extension();

  static ImmutableMedicinalProduct_NamePart.TypeBuildStage builder() {
    return ImmutableMedicinalProduct_NamePart.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/MedicinalProduct_NamePart.ts
import { Coding } from "./Coding";
import { Extension } from "./Extension";

export interface MedicinalProduct_NamePart {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    part: string | null | undefined;
    type: Coding;
}
